packages/angular_devkit/schematics/src/engine/engine.ts

```diff
@@ -140,4 +162,42 @@ export class SchematicEngine<CollectionT extends object, SchematicT extends obje
         return hostSource;
     }
   }
+
+  createTaskExecutor(name: string): Promise<TaskExecutor> {
+    // Hard coded tasks with default factory options for now
+    switch (name) {
+      case 'npm':
+        const location = '../../tasks/npm_task';
+
+        return import(location).then(mod => mod.default());
+    }
+
+    return Promise.reject(new Error(`Unknown task '${name}' requested.`));
+  }
+
+  executeTasks(): Observable<void> {
+    const executors = new Map<string, TaskExecutor>();
+
+    const taskObservable = Observable
+      .from(this._taskSchedulers)
```

Comment:
We use lettable operators now because it's where RxJS is going.

Reply:
Done.